core/src/main/scala/org/dbpedia/extraction/wikiparser/NodeUtil.scala

```diff
@@ -210,4 +213,32 @@ object NodeUtil
 
         splitNodes.reverse
     }
+
+    def removeEmptyTextNode(list : List[Node]) : List[Node] = {
+        var tempList = ListBuffer[Node]()
+
+        for( node <- list){
+            if(notEmpty(node)){
+                tempList += node
+            }
+        }
+        tempList.toList
+    }
+
+    def notEmpty(node : Node) : Boolean = {
+
+        if(!node.isInstanceOf[TextNode]){
+            return true;
+        }
+
+        var content = node.asInstanceOf[TextNode].text.trim
+
+        if (content == "" || content == "\n"){
```

Comment:
`trim()` removes any whitespace like `\n`
something like `return ! node.asInstanceOf[TextNode].text.trim.empty` should work